### PR TITLE
docs: add content to README.md

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

The upfadapter README.md was empty. Here we add some content to make it easier for developers and users to understand what `upfadapter` is.

### CODE REVIEW

**Review by gab-arrobo — COMMENTED** [review]
@gruyaume,
FYI, the `upfadapter` was nested into the `smf` repo. But, since this is a separate component/pod, we moved it to its own repo.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
+SPDX-FileCopyrightText: 2024 Canonical Ltd.
 SPDX-License-Identifier: Apache-2.0
 -->
 <!-- [![Go Report Card](https://goreportcard.com/badge/github.com/omec-project/upfadapter)](https://goreportcard.com/report/github.com/omec-project/upfadapter) -->
 
 # upfadapter
+
+`upfadapter` is required to scale the SMF's in SD-Core. The SMF sends custom PFCP messages to `upfadapter`, which intercepts and modifies specific fields before sending the messages to the UPF. 
+
+![alt text](image.png)
```
[the remark this answers is not in the document]
Comment (gruyaume) [reply README.md:12]: My bad, it's here now

File: image.png
Comment (gab-arrobo) [inline image.png:1]: You need to upload another file `image.png.license` for the copyright and license header. You can use the `go.mod.license` as a reference

**Review by gab-arrobo — APPROVED** [review]
+1

**Pull request merged**